official/wasm/pkg/config/config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/pkg/errors"
+)
+
+type (
+	// options represents configuration for the wasm config.
+	options struct {
+		simulationGasLimit uint64
+		smartQueryGasLimit uint64
+		memoryCacheSize    uint32
+	}
+	// Option configures the message scaffolding.
+	Option func(*options)
+)
+
+// newOptions returns a wasmOptions with default options.
+func newOptions() options {
+	return options{
+		simulationGasLimit: 0,
+		smartQueryGasLimit: 3_000_000,
+		memoryCacheSize:    100,
+	}
+}
+
+// WithSimulationGasLimit provides a simulation gas limit for the wasm config.
+func WithSimulationGasLimit(simulationGasLimit uint64) Option {
+	return func(m *options) {
+		m.simulationGasLimit = simulationGasLimit
+	}
+}
+
+// WithSmartQueryGasLimit provides a smart query gas limit for the wasm config.
+func WithSmartQueryGasLimit(smartQueryGasLimit uint64) Option {
+	return func(m *options) {
+		m.smartQueryGasLimit = smartQueryGasLimit
+	}
+}
+
+// WithMemoryCacheSize provides a memory cache size for the wasm config.
+func WithMemoryCacheSize(memoryCacheSize uint32) Option {
+	return func(m *options) {
+		m.memoryCacheSize = memoryCacheSize
+	}
+}
+
+// AddWasm add wasm parameters to the chain TOML config.
+func AddWasm(configPath string, options ...Option) error {
+	// Create the options.
+	opts := newOptions()
+	for _, apply := range options {
+		apply(&opts)
+	}
+
+	f, err := os.OpenFile(configPath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0o600)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	// Check if the wasm section already exist.
+	if hasWasm(configPath) {
+		return errors.Errorf("config file already have wasm %s", configPath)
+	}
+
+	// Add default configs.
+	config := wasmtypes.DefaultWasmConfig()
+	if opts.simulationGasLimit != 0 {
+		config.SimulationGasLimit = &opts.simulationGasLimit
+	}
+	config.SmartQueryGasLimit = opts.smartQueryGasLimit
+	config.MemoryCacheSize = opts.memoryCacheSize
+
+	// Save new configs to the TOML file.
+	wasmConfig := fmt.Sprintf("%s", wasmtypes.ConfigTemplate(config))
```

Comment:
I believe is not necessary to wrap the `ConfigTemplate` call in an `fmt.Sprintf`, maybe be a leftover?